modelci/cli/modelhub.py

```diff
@@ -242,3 +244,194 @@ def delete(model_id: str = typer.Argument(..., help='Model ID')):
+        dtype_output: Optional[str] = typer.Option(None, '-do', '--dtypeo', help='output data type such as \'float\''),
+
+        model_task: Optional[str] = typer.Option(..., '-t', '--task', help='task of model'
+                                                                           'such as \'IMAGE_CLASSIFICATION\''
+                                                                           'or \'SEGMENTATION\'\'OBJECT_DETECTION\'')
+):
+    """
+    TODO replace input and output str with IOShape
+    """
+    import modelci.hub.converter.converter as cvt
+    way = (src, dst)
+    task = {
+        'IMAGE_CLASSIFICATION': saveTask.IMAGE_CLASSIFICATION,
+        'SEGMENTATION': saveTask.SEGMENTATION,
+        'OBJECT_DETECTION': saveTask.OBJECT_DETECTION
+    }
+    in_format = {
+            'FORMAT_NONE': ModelInputFormat.FORMAT_NONE,
+            'FORMAT_NHWC': ModelInputFormat.FORMAT_NHWC,
+            'FORMAT_NCHW': ModelInputFormat.FORMAT_NCHW,
+    }
+    if way == ('keras', 'onnx'):
+        import onnx
+        import tensorflow as tf
+        loaded = tf.saved_model.load(model_path)
+        save_path = generate_path(model_name=model_name,
+                                  framework=saveFramework.TENSORFLOW,
+                                  task=task[model_task],
+                                  engine=saveEngine.ONNX,
+                                  version=version)
+        save_path.parent.mkdir(parents=True, exist_ok=True)
+        onnx_model = cvt.convert(model=loaded, src_framework='keras', dst_framework='onnx')
+        onnx.checker.check_model(onnx_model)
+        onnx.save(onnx_model, save_path)
+    if way == ('tensorflow', 'onnx'):
+        import onnx
+        save_path = generate_path(model_name=model_name,
+                                  framework=saveFramework.TENSORFLOW,
+                                  task=task[model_task],
+                                  engine=saveEngine.ONNX,
+                                  version=version)
+        save_path.parent.mkdir(parents=True, exist_ok=True)
+        onnx_model = cvt.convert(model=model_path, src_framework='tensorflow', dst_framework='onnx')
+        onnx.checker.check_model(onnx_model)
+        onnx.save(onnx_model, save_path)
+    if way == ('keras', 'tensorflow_serving'):
+        import tensorflow as tf
+        save_path = generate_path(model_name=model_name,
+                                  framework=saveFramework.TENSORFLOW,
+                                  task=task[model_task],
+                                  engine=saveEngine.TFS,
+                                  version=version)
+        save_path.parent.mkdir(parents=True, exist_ok=True)
+        loaded = tf.saved_model.load(model_path)
+        cvt.convert(model=loaded, src_framework='tensorflow', dst_framework='tfs', save_path=save_path)
+    if way == ('xgboost', 'onnx'):
```

Comment:
If I were you, I would write something like 

```python
if way[0] == 'xgboost':
    import onnx
    import torch
    save_engine = saveEngine.PYTORCH
    save_framework = saveFramework.PYTORCH
    if is_onnx:
        save_framework = saveFramework.ONNX
        save_engine = saveEngine.ONNX
        
    loaded = torch.load(model_path)
    inputs = [convertIO(shape=eval(shape_input), dtype=eval(dtype_input), name='input_0')]
    onnx_model = cvt.convert(model=loaded, src_framework='xgboost', dst_framework='onnx', inputs=inputs)
    onnx.checker.check_model(onnx_model)
    save_path = generate_path(model_name=model_name, framework=save_framework,
                            task=task[model_task], engine=save_engine, version=version)
    save_path.parent.mkdir(parents=True, exist_ok=True)

    if is_onnx:
        onnx.save(onnx_model, save_path)
    else:
        torch.save(torch_model, save_path)
```

Try to merge some if-elses together to reduce the code redundancy.

Reply:
Thank you very much, I am working on this right now